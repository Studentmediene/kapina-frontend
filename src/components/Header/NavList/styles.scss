@import 'variables.scss';

.navbarList {
  flex: 1;
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 0;
}

.navbarItem {
  list-style: none;
  display: block;
  font-size: 21px;
}

.navbarItem:hover {
  text-decoration: underline;
}

.navbarLink {
  text-decoration: none;
  color: inherit;
}

@media screen and (min-width: $breakpoint-medium) {
  .navbarItem {
    padding: 0px 20px;
  }
}