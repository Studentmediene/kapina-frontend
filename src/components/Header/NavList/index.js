import React from 'react';
import { Link } from 'react-router-dom';

import styles from './styles.scss';

const navbarComponents = links => links.map(link => (
  <li key={link.path} className={styles.navbarItem}>
    <Link className={styles.navbarLink} to={link.path}>
      {link.title}
    </Link>
  </li>
))

export const NavList = (props) => (
  <ul className={styles.navbarList}>
    {navbarComponents(props.links)}
  </ul>
);

export default NavList;
