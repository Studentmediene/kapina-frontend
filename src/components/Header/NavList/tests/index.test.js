import React from 'react';
import { shallow } from 'enzyme';

import NavList from '../';

describe('<NavList />', () => {
  it('renders correctly', () => {
    const props = {
      links: [
        {
          path: '/link1',
          title: 'Link1',
        },{
          path: '/link2',
          title: 'Link2',
        }
      ]
    }
    const tree = shallow(<NavList {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
