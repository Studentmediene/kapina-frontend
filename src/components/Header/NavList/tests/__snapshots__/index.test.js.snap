// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavList /> renders correctly 1`] = `
<div
  className="navbar"
>
  <ul
    className="navbarList"
  >
    <li
      className="navbarItem"
      key="/link1"
    >
      <Link
        className="navbarLink"
        replace={false}
        to="/link1"
      >
        Link1
      </Link>
    </li>
    <li
      className="navbarItem"
      key="/link2"
    >
      <Link
        className="navbarLink"
        replace={false}
        to="/link2"
      >
        Link2
      </Link>
    </li>
  </ul>
</div>
`;
